crates/rpc/src/lib.rs

```diff
@@ -18,6 +20,31 @@ impl BeerusRpc {
     }
 
     pub async fn run(self) -> Result<(SocketAddr, ServerHandle), Error> {
+        let local_spec_version =
+            self.spec_version().await.expect("Should never happen");
+
+        let remote_spec_version = self
+            .beerus
+            .starknet_client
+            .spec_version()
+            .await;
+
+        // check remote spec version
+        match remote_spec_version {
+            Ok(remote_version) => {
+                if remote_version != local_spec_version {
+                    error!(
```

Comment:
Duplicated error message (DRY). We can just return an error and let client code decide if it should be logged or not.